sdk/nodejs/common/errors/DockerImageRefValidationError.ts

```diff
@@ -10,7 +10,7 @@ interface DockerImageRefValidationErrorOptions extends DaggerSDKErrorOptions {
  */
 export class DockerImageRefValidationError extends DaggerSDKError {
   name = "DockerImageRefValidationError"
-  code = "D104"
+  code = "D107"
```

Comment:
We are using codes but none of them are documented. 
We should probably document it above by adding a `code` with an explanation of what's happening.

Reply:
Sorry if I was not clear enough. What I meant is we are using codes but what is their purpose if they are not documented ?
If I remember our initial idea was to get something like this: 
https://www.prisma.io/docs/reference/api-reference/error-reference#error-codes
So either we can get rid of them because we feel that classes names + message give enough input, or we use JSdocs to document them. WDYT ?

Reply:
Right.
I think we are missing something though. Like creating consts with all of those Dxxx codes into a file like the this:

```js
/*
* @param D100
* @description The dagger specific error code. Use this to identify dagger errors programmatically.
*/
export const D100 = "D100"

/*
* @param D101
* @description The dagger specific error code. Use this to identify dagger errors programmatically.
*/
export const D101 = "D101"

...
```

What I see as a benefit is having every errors code centralized that will let us what is the next Dxxx code available. And we can also get a dedicated page with error code instead of browsing every errors class.
For instance we are currently refactoring the SDK and it will imply new errors handlers. So I need to open every errors class to check which is the next error code available, does it make sense ?

Reply:
Yeah this is a good idea! I will add this to the PR if wanted.

Reply:
Cool :)! If you have no time I can also do it. Let me know. Thanks

Reply:
Perfect! thanks @benjaminstrasser